sdk/storage/Azure.Storage.Common/src/Errors.cs

```diff
@@ -60,6 +60,9 @@ public static ArgumentException InvalidPermission(char s)
         public static ArgumentException InvalidResourceType(char s)
             => new ArgumentException($"Invalid resource type: '{s}'");
 
+        public static ArgumentException ArgumentNotSupported(string paramName)
+           => new ArgumentException("Argument not supported", paramName);
```

Comment:
we need a more specific message here like `The service version is not supported`